TrafficSimulator/Assets/Utilities/DayNightManager.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Simulation
+{
+    public enum SunMode
+    {
+        Time,
+        Day,
+        Night
+    }
+
+    public class DayNightManager : MonoBehaviour
+    {
+        public SunMode _sunMode = SunMode.Time;
+
+        void Start()
+        {
+            SetSunLocation(_sunMode);
+
+            // Subscribe to time manager events for hour
+            TimeManager.OnHourChanged += MoveSunByTime;
+        }
+
+        private void SetSunLocation(SunMode mode)
+        {
+            switch (mode)
+            {
+                case SunMode.Time:
+                    break;
+                case SunMode.Day:
+                    transform.rotation = Quaternion.Euler(90, 0, 0);
+                    break;
+                case SunMode.Night:
+                    transform.rotation = Quaternion.Euler(-90, 0, 0);
+                    break;
+            }
+        }
+
+        private void MoveSunByTime()
```

Comment:
This needs to be changed. This function should tick the sun forward each hour, currently it overrides the rotation. This means that regardless of the `sunMode`, the sun will always follow the time (once an hour passes so this function is called).

Instead it should read the current rotation, and move it forward one tick (1/24th) each hour, and handle the edge cases (sunrise and sunset)